mini-kodein-android/src/main/java/mini/kodein/android/KodeinAndroidUtils.kt

```diff
@@ -51,75 +47,85 @@ class KodeinViewModelFactory(private val injector: DKodein,
 }
 
 /**
- * Injects a [ViewModel] into a [FragmentActivity] that implements [KodeinAware].
+ * Injects a [ViewModel] into a [FragmentActivity] that implements [DIAware].
  */
 @MainThread
-inline fun <reified VM : ViewModel, A> A.viewModel(): Lazy<VM> where A : KodeinAware, A : FragmentActivity {
+inline fun <reified VM : ViewModel, A> A.viewModel(): Lazy<VM> where A : DIAware, A : FragmentActivity {
     return lazy {
         ViewModelProvider(this, direct.instance()).get(VM::class.java)
     }
 }
 
 /**
- * Injects a [ViewModel] into a [Fragment] that implements [KodeinAware].
+ * Injects a [ViewModel] into a [Fragment] that implements [DIAware].
  */
 @MainThread
-inline fun <reified VM : ViewModel, F> F.viewModel(): Lazy<VM> where F : KodeinAware, F : Fragment {
+inline fun <reified VM : ViewModel, F> F.viewModel(): Lazy<VM> where F : DIAware, F : Fragment {
     return lazy {
         ViewModelProvider(this, direct.instance()).get(VM::class.java)
     }
 }
 
 /**
- * Injects a [ViewModel] into a [FragmentActivity] that implements [KodeinAware].
+ * Injects a [ViewModel] into a [FragmentActivity] that implements [DIAware].
  *
  * Requires previous [ViewModelProvider.Factory] injection for the [ViewModel] via [bindViewModelFactory]
  * to work and a [TypedViewModel] to be used.
  */
-@MainThread
-inline fun <reified T, reified VM : TypedViewModel<T>, A> A.viewModel(params: T): Lazy<VM> where A : KodeinAware, A : FragmentActivity {
-    return lazy {
-        ViewModelProvider(this, direct.instance(VM::class.java, params)).get(VM::class.java)
-    }
-}
+// FIXME: This is failing after the migration to Kodein 7.1.0. It seems not being properly aware that
+//        T is reified, so [params] doesn't need to be a [Typed<TypedValue(T)>] and that it can be
+//        directly of type T
+//@MainThread
+//inline fun <reified T, reified VM : TypedViewModel<T>, A> A.viewModel(params: T): Lazy<VM> where A : DIAware, A : FragmentActivity {
+//    return lazy {
+//        ViewModelProvider(this, direct.instance(VM::class.java, params)).get(VM::class.java)
+//    }
+//}
```

Comment:
Did you try this?
```suggestion
@MainThread
inline fun <reified T : Any, reified VM : TypedViewModel<T>, A> A.viewModel(params: T): Lazy<VM> where A : DIAware, A : FragmentActivity {
    return lazy {
        ViewModelProvider(this, direct.instance(VM::class.java, params)).get(VM::class.java)
    }
}
```
Just make `T` extend from `Any` and it should work :P